File: solutions/2023/7-Camel_Cards/s1.js
const utils = require('../../../language-setups/javascript/utils.js')
const path = require('path')
const readFileSync = require('fs').readFileSync
const inputPath = path.join(__dirname, 'input_example')
const data = readFileSync(inputPath, 'utf8')

parseInput = (data) => {
    const lines = data.split(/\n/)

   const  handBets = lines.map(line=>line.split(' ')).map(handBet=>[handBet[0],Number(handBet[1])])

    return handBets
}

// strong -> weak
const handStrength = ["five", "four", "full", "three", "two", "one", "high"]
const cardStrength = ["A", "K", "Q", "J", "T", "9", "8", "7", "6", "5", "4", "3", "2"]

getHandType = (hand) => {
// assumes hand is length 5 and valid

    if (/(.)\1{4}/.test(hand)) {
        return 'five'
    } else if (/(.).*\1.*\1.*\1/.test(hand)) {
        return 'four'
    } else if (//.test(hand)) {
        return 'full'
    } else if () {
        return 'three'
    } else if () {
        return 'two'
    } else if () {
        return 'one'
    } else {
        return 'high'
    }
}

const handBets = parseInput(data)
console.log(handBets)

// sort by strength of hand weak->strong
// multiply index+1 by bet
// sum









const SOLUTION_1 = undefined

console.log(`SOLUTION_1: ${SOLUTION_1}`)
// Example solution: 6440

module.exports = {parseInput, getHandType}

File: solutions/2024/7-Bridge_Repair/s1.py
import os
current_dir = os.path.dirname(os.path.realpath(__file__))
# input_path = os.path.join(os.path.dirname(__file__), 'input_example')
input_path = os.path.join(os.path.dirname(__file__), 'input')

with open(input_path, 'r', encoding='utf8') as file:
    data = file.read().split('\n')
    
def solution_1(data):    
    solution = 0
    
    for line in data:
        result = int(line.split(':')[0])
        nums = [int(_) for _ in line.split(': ')[1].split(' ')]
        
        index = 0
        
        possible_results = [nums[0]]
        
        while index < len(nums) - 1:
            index += 1
            
            new_results_add = [
                _ + nums[index]
                for _ in possible_results
            ]
            new_results_mult = [
                _ * nums[index]
                for _ in possible_results
            ]
            
            possible_results = new_results_add + new_results_mult
            
        if result in possible_results:
            solution += result
            
    print(solution)
        
# solution_1(data)

def solution_2(data):    
    solution = 0
    
    for line in data:
        result = int(line.split(':')[0])
        nums = [int(_) for _ in line.split(': ')[1].split(' ')]
        
        index = 0
        
        possible_results = [nums[0]]
        
        while index < len(nums) - 1:
            index += 1
            
            new_results_add = [
                _ + nums[index]
                for _ in possible_results
            ]
            new_results_mult = [
                _ * nums[index]
                for _ in possible_results
            ]
            new_results_concat = [
                int(f'{_}{nums[index]}')
                for _ in possible_results
            ]
            
            possible_results = new_results_add + new_results_mult + new_results_concat
            
        if result in possible_results:
            solution += result
            
    print(solution)
        
solution_2(data)

